Adding median absolute deviation documentation page (#6808)
opensearch-project/documentation-website: _aggregations/metric/median-absolute-deviation.md

---
layout: default
title: Median absolute deviation
parent: Metric aggregations
grand_parent: Aggregations
nav_order: 65
redirect_from:
  - /query-dsl/aggregations/metric/median-absolute-deviation/
---

# Median absolute deviation aggregations

> The `median_absolute_deviation` metric is a single-value metric aggregation that returns median absolute deviation field. Median absolute deviation is a statistical measure of data variability. It is used to measure dispersion from the median, may be less impacted by outliers in a dataset. 

Reviewer: ```suggestion
The `median_absolute_deviation` metric is a single-value metric aggregation that returns median absolute deviation field. Median absolute deviation is a statistical measure of data variability. Since the median absolute deviation measures dispersion from the median, it provides a more robust measure of variability that is less affected by outliers in a dataset. 
```

Median absolute deviation is calculated with:<br>
median_absolute_deviation = median(|X<sub>i</sub> - Median(X<sub>i</sub>)|)

> The following example calculates the median absolute deviation of the `DistanceMiles` field of the opensearch_dashboards_sample_data_flights:

Reviewer: ```suggestion
The following example calculates the median absolute deviation of the `DistanceMiles` field of the sample dataset `opensearch_dashboards_sample_data_flights`:
```

```json
GET opensearch_dashboards_sample_data_flights/_search
{
  "size": 0,
  "aggs": {
    "median_absolute_deviation_DistanceMiles": {
      "median_absolute_deviation": {
        "field": "DistanceMiles"
      }
    }
  }
}
```
{% include copy-curl.html %}

#### Example response

```json
{
  "took": 35,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 10000,
      "relation": "gte"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "median_absolute_deviation_distanceMiles": {
      "value": 1829.8993624441966
    }
  }
}
```

### Missing

> You can set a default value for missing fields from documents by specifying the `missing` parameter. This could be a missing field or a null value in a field.

Reviewer 2: Should we just rephrase this to:

```
By default, a missing field in or a null value in a field from a document are ignored in computations. You can specify a value for them by specifying the `missing` parameter. 
```

I don't like the idea of saying that we are providing a default value. It is more like treat the missing values as this value which I'm specifying.

> You can set a default value for missing fields from documents by specifying the `missing` parameter. This could be a missing field or a null value in a field.

Reviewer: ```suggestion
You can set a default value for fields that are missing from documents by specifying the `missing` parameter. The value can be `missing` or `null`.
```

```json
GET opensearch_dashboards_sample_data_flights/_search
{
  "size": 0,
  "aggs": {
    "median_absolute_deviation_distanceMiles": {
      "median_absolute_deviation": {
        "field": "DistanceMiles",
        "missing": 1000
      }
    }
  }
}
```
{% include copy-curl.html %}

#### Example response

```json
{
  "took": 7,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 10000,
      "relation": "gte"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "median_absolute_deviation_distanceMiles": {
      "value": 1829.6443646143355
    }
  }
}
```

### Compression

> The calculation of the median absolute deviation utilizes [t-digest](https://github.com/tdunning/t-digest/tree/main) which controls the balance between performance and accuracy of estimation. The default value for TDigest's `compression` field is 1000. Decreasing the `compression` value will increase the performance while reducing the accuracy of the estimation.

Reviewer 2: I think we need to add in a little (very-little) detail on `compression` here. 
  
  
How about:

-----
The calculation of the median absolute deviation utilizes [t-digest](https://github.com/tdunning/t-digest/tree/main) data-structure which controls the balance between performance and accuracy of estimation. T-Digest controls this balance through `compression` parameter, with a default value of 1000. Adjusting the `compression` value affects the trade-off between computational efficiency and the precision of the estimation. Lowering the compression value improves performance but may lead to slightly less accurate results, while increasing it enhances accuracy at the cost of increased computational overhead.

----

> The calculation of the median absolute deviation utilizes [t-digest](https://github.com/tdunning/t-digest/tree/main) which controls the balance between performance and accuracy of estimation. The default value for TDigest's `compression` field is 1000. Decreasing the `compression` value will increase the performance while reducing the accuracy of the estimation.

Reviewer: ```suggestion
The median absolute deviation calculation uses [t-digest](https://github.com/tdunning/t-digest/tree/main), which controls the balance between performance and accuracy of estimation. The default value for the t-digest `compression` field is `1000`. Decreasing the `compression` value increases the performance while reducing the accuracy of estimation.
```
